starknet_py/tests/e2e/docs/guide/test_declaring_contracts.py

```diff
@@ -1,13 +1,17 @@
+import sys
+
 import pytest
 
 from starknet_py.net.client_models import ResourceBounds
 
 
-# TODO (#1419): Fix contract redeclaration
-@pytest.mark.skip(reason="Redeclaration occurred")
+@pytest.mark.skipif(
+    "--contract_dir=v2" not in sys.argv,
+    reason="Contract exists only in v2 directory",
+)
 @pytest.mark.asyncio
-async def test_declaring_contracts(account, map_compiled_contract_and_class_hash):
-    (compiled_contract, class_hash) = map_compiled_contract_and_class_hash
+async def test_declaring_contracts(account, map_compiled_contract_and_class_hash_copy1):
```

Comment:
nit: Also please apply it to other variables 🙏 .
```suggestion
async def test_declaring_contracts(account, map_compiled_contract_and_class_hash_copy_1):
```